src/momento/auth/momento_endpoint_resolver.py

```diff
@@ -11,27 +14,53 @@
 _CACHE_ENDPOINT_CLAIM_ID = "c"
 
 
-class _Endpoints:
-    def __init__(self, control_endpoint: str, cache_endpoint: str):
-        self.control_endpoint = control_endpoint
-        self.cache_endpoint = cache_endpoint
+@dataclass
+class _TokenAndEndpoints:
+    control_endpoint: str
+    cache_endpoint: str
+    auth_token: str
 
 
-def resolve(auth_token: str, endpoint_override: Optional[str] = None) -> _Endpoints:
-    if endpoint_override is not None:
-        return _Endpoints(
-            _MOMENTO_CONTROL_ENDPOINT_PREFIX + endpoint_override,
-            _MOMENTO_CACHE_ENDPOINT_PREFIX + endpoint_override,
+@dataclass
+class Base64DecodedToken:
```

Comment:
i think you want v1 in this name in case there is a v2 later.  also @bruuuuuuuce do we have an easy way to differntiate those in the future?  presumably there is a version field we are just ignoring for now?

also should this class have an underscore prefix?

Reply:
Good call on both. Will rename.